main.nf

```diff
@@ -130,18 +112,13 @@ if (params.protocol == "illumina"){
 if( !params.mature || !params.hairpin ){
     exit 1, "Missing mature / hairpin reference indexes! Is --genome specified?"
 }
-if( params.mature ){
-    mature = file(params.mature)
-    if( !mature.exists() ) exit 1, "Mature file not found: ${params.mature}"
-}
-if( params.hairpin ){
-    hairpin = file(params.hairpin)
-    if( !hairpin.exists() ) exit 1, "Hairpin file not found: ${params.hairpin}"
-}
-if( params.gtf ){
-    gtf = file(params.gtf)
-    if( !gtf.exists() ) exit 1, "GTF file not found: ${params.gtf}"
-}
+
+if (params.mature) { mature = file(params.mature, checkIfExists: true) } else { exit 1, "Mature file not found: ${params.mature}" }
```

Comment:
Are these mandatory parameters? If not, then you dont need the `else` statement. See https://github.com/drpatelh/chipseq/blob/ee6456ca469a43521fe41551cf9f4f5c527a5d0e/main.nf#L159

Reply:
the GTF is not, but other two, they are.I'll update that.